app/controllers/replies_controller.rb

```diff
@@ -297,19 +300,147 @@ def require_edit_permission
     redirect_to post_path(@reply.post)
   end
 
-  def preview(written)
-    @written = written
-    @post = @written.post
+  def get_multi_replies
+    @multi_replies_json = JSON.parse(params.fetch(:multi_replies_json, "[]"))
```

Comment:
Maybe we can sanitize `@multi_replies_json` here (put it through `permitted_params`, separate before doing the `Reply.new`) and then assume in subsequent sections that it's safe